vignettes/pr_review_guidance.Rmd

```diff
@@ -0,0 +1,97 @@
+---
+title: "Pull Request Review Guidance"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{pr_review_guidance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This document is intended to be a guidance for reviewers of pull requests (PRs) in the {admiral} package. PR authors will benefit from shorter review times by closely following the guidance provided here.
+
+
+# Review Criteria
+
+For a pull request to be merged it needs to pass the automated `R CMD check` and `lintr` workflows on GitHub at a minimum. These checks can also be run locally using the `devtools::check()` and `lintr::lint_package()` commands, respectively. In addition the PR reviewer should make sure that
+
+- the [Programming Strategy](programming_strategy.html) is followed
+
+- the function is ADaM IG compliant
+
+- the function does what is intended for (as described in the header and corresponding issue)
+
+- the function header properly explains the function
+
+- the implemented derivation is in the scope of admiral, e.g. does not expect company specific input or hard-code company-specific rules
+
+- meaningful error or warning messages are issued if the input is invalid
+
+- documentation is created/updated by running `devtools::document()`
+
+- functions which are supposed to be exported are listed in the `NAMESPACE` file; this requires an `@export` tag in the function header
+
+- examples print relevant source variables and newly created variables and/or records in their output
+
+- the `NEWS.md` file is updated with an entry that explains the new features or changes
+
+- the author of a function is listed in the `DESCRIPTION` file
+
+- all files affected by the implemented changes, e.g. vignettes and templates, are updated
+
+
+# Common R CMD Check Issues
+
+`R CMD check` is a command line tool that checks R packages against a standard set of criteria. For a pull request to pass the check must not issue any notes, warnings or errors. Below is a list of common issues and how to resolve them.
+
+## Package Dependencies
+
+```
+> checking package dependencies ... ERROR
+  Namespace dependency not required: ‘pkg’
+```
+
+Add `pkg` to the `Imports` or `Suggests` field in the `DESCRIPTION` file. In general, dependencies should be listed in the `Imports` field. However, if a package is only used inside vignettes or unit tests it should be listed in `Suggests` because all {admiral} functions would work without these "soft" dependencies being installed.
+
```

Comment:
I would add how the checks can be executed locally using the same versions as github. This is especially helpful if something is failing just for some versions.